### PR TITLE
Refactor `StandaloneLedger`

### PR DESCRIPTION
**Pull request by AlexTheWizardL**
### Refactor `StandaloneLedger` to Remove Class Variables

This PR refactors `StandaloneLedger` by removing all class variables to ensure it functions purely as an abstract interface, without direct data storage or implementations for accessors or mutators.

**Changes in this PR:**
- Removed all class variables from `StandaloneLedger`.
- Introduced `_prices_as_dict_of_df`, aligning with the approach used for `_assets_as_dict_of_df` to optimize data handling.
- Revised the `complete_ledger()` method to eliminate local variable dependencies.

**Note:** The caching strategy remains challenging, particularly for cache invalidation across multiple elements (e.g., tax, prices). This PR includes an my input for discussion on how to best handle caching in this context. Feedback and ideas on improving cache management and invalidation would be highly appreciated.

### CODE REVIEW

File: pyledger/standalone_ledger.py [hunk 1]
```diff
@@ -339,11 +141,10 @@ def complete_ledger(self) -> None:
             date=df.loc[index, "date"]
         )
 
-        # revaluations
+        # Revaluations
         revaluations = self.revaluations()
         reporting_currency = self.reporting_currency
         for row in revaluations.to_dict("records"):
-            self._serialized_ledger = self.serialize_ledger(df)
```
Comment (lasuk) [inline pyledger/standalone_ledger.py:346]: With this change, `account_balance(..)` further below will return a different value.
I need to sleep over this before making a proposal.

File: pyledger/standalone_ledger.py [hunk 2]
```diff
@@ -438,7 +235,7 @@ def _report_amount(
             raise ValueError("Vectors 'amount', 'currency', and 'date' must have the same length.")
         result = [
             self.round_to_precision(
-                a * self.price(t, date=d, currency=reporting_currency)[1],
+                a * 100,
```
Comment (lasuk) [inline pyledger/standalone_ledger.py:238]: I do not understand this change.

**Review by lasuk — COMMENTED** [review]
Thanks for converting StandaloneLedger to an abstract class.

We can delete even more unneeded code: I've added four more commits to drop or reshuffle methods.

I've left two review comments to discuss.

File: pyledger/standalone_ledger.py (hunk 1 above)
Comment (lasuk) [reply pyledger/standalone_ledger.py:346]: Fixed in 5dcbb2e:

- Compute account balance from ledger entries provided as argument to `_balance_from_serialized_ledger`.
- Isolate computation of revaluation entries

File: pyledger/standalone_ledger.py (hunk 2 above)
Comment (AlexTheWizardL) [reply pyledger/standalone_ledger.py:237]: This was made by a mistake. Should be rollbacked

Comment (AlexTheWizardL) [reply pyledger/standalone_ledger.py]: UPD: rollbacked

File: pyledger/standalone_ledger.py (hunk 1 above)
Comment (lasuk) [reply pyledger/standalone_ledger.py:346]: This has been implemented and tested in the meantime.

**Comment by lasuk** [comment]
Merging. Thanks for the excellent work!

**Pull request merged**